src/librustc_mir/interpret/terminator.rs

```diff
@@ -293,6 +293,31 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     }
                 }
 
+                // If this function is a const function then as an optimisation we can query this
```

Comment:
This function looks pretty big -- seems like this bit is something we can extract out to a new one.

Reply:
Thanks, done so!